### DISCUSSION TITLE
Mark monitor as inactive

### DISCUSSION THREAD

**Discussion by WolfgangFahl** [rxhanson/Rectangle#457] [Ideas]
Rectangle is an exccellent solution for my usecase. I am happy that i found it via an [apple.stackexchange question](https://apple.stackexchange.com/questions/28569/is-there-a-keyboard-shortcut-to-move-a-window-from-one-monitor-to-another)

In my setup i am using 4 screens. Three of them are side by side on the desktop the 4th one is a projector that i only use occasionaly. When I am holding trainings the setup is different and i have two projectors and two desktop screens. 

In the different setups it would be good to mark a monitor/screen as "inactive". As soon as doing so all windows from that screen should be moved to the other screens. E.g. when switching off the projector the windows should move to the desktop screen.  

There are lot of followup ideas on this usecase e.g. restoring the windows layout for a given setting. Have such ideas already been discussed here?

**Comment by rxhanson** [comment]
Sorry for the delayed response. This idea makes sense to me. It's a little bit on the outer edge of what I would be looking to add into Rectangle, but I would consider a PR for it and can give direction on where to start looking in the code if anyone wants to go down that path.

**Comment by WolfgangFahl** [comment]
@rxhanson thx for the response. Currently my 4 screen MacPro is in repair and i'll have to wait a few weeks until i get it back. Unfortunately I am not familiar with swift and would therefore have a hard time contributing. Still I might give it a try if you let me know what a starting point might be.

**Reply by rxhanson** [reply]
The best point to start is actually understanding Todo mode (a feature created by the CEOs of GitHub & Stripe), as your feature would borrow a lot of the same elements. Here's the wiki page on Todo mode: https://github.com/rxhanson/Rectangle/wiki/Todo-Mode. In the code, you'll probably want to start with the `TodoManager`. 

There you can walk a path of looking at moving all windows on a display, in the `moveAll` function, and work your way out to see how ScreenDetection works, and ultimately adding a menu bar menu item for selecting a display.
